### DISCUSSION TITLE
Will Audio MIDI setup be included

### DISCUSSION THREAD

**Discussion by Forrhouse** [ravynsoft/ravynos#269] [General]
I have several important questions:

1) Wondering if an "Audio MIDI setup" like interface is and or will be included in this OS? Keeping the structure / Page and Menu 
    Layout the same and keeping the way it works the same along with any improvements would be very Important.  Including the 
    ability to cascade Audio and MIDI interfaces is also extremely important. For me personally, and a lot of other people, this is very 
    important.  I believe a lot of people would start using this OS. Not having this functionality would deter many from using it.

2) Will MIDI 2.0 be included as in OS Big Sur and OS Monterey?

3) Will a Core MIDI and Core Audio equivalent be included and will it be compatible with the one in Mac Os?

4) Lastly, will VST 1 through VST 3, Audio Unit and AAX plugin format compatibility be included?

These are all things that Pro Audio types will want and which would be deal breakers for professionals and hobbyists alike if they are not.

A lot of people will want Apples "Logic" DAW software to work in it as well.

**Comment by mszoek** [comment]
Hi! :)

So, the goal is to have all of this stuff supported and compatible with macOS Catalina as our initial target. We'll evolve APIs and feature support from there. Pro Audio and music comp is not the top priority right now as I'm focused on much more basic things, but those topics are near to my heart and definitely on the list!

The current idea for CoreAudio is to wrap its APIs around [JACK](https://jackaudio.org/) over [FreeBSD's pcm audio](https://www.freebsd.org/cgi/man.cgi?query=pcm&sektion=4&format=html) which is quite capable. This should provide very similar capabilities and low latency.

Not sure yet of the approach for CoreMIDI.

**Reply by Forrhouse** [reply]
I was wondering if this would help when implementing a Core Audio Equivalent: 
https://rnine.github.io/SimplyCoreAudio/Classes/SimplyCoreAudio.html

Also found these that may be helpful:

https://github.com/tpoechtrager/osxcross
(The goal of OSXCross is to provide a well working macOS cross toolchain for Linux, FreeBSD, OpenBSD, and Android [Termux].)

https://github.com/RustAudio

Also, the lower latency the better. If you can beet the latency in Mac OS then you would probably have even more converts.

I read something about having an OSs kernel being optimized for low latency being important for Audio apps.  Will this be the case for  RavynOs?  I'm not a coder, so please excuse any ignorance.

Support for multi-client ASIO implementation would be nice! (ASIO playing in more than one program)
